### PR TITLE
fix(module:cascader): update position when click menu item

### PR DESCRIPTION
**Pull request by yutao331763646**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by yutao331763646** [comment]
image
与antd表现不一致

**Review by wendellhu95 — APPROVED** [review]
LGTM. Thanks!

**Comment by yutao331763646** [comment]
目前的问题，如下图，当下拉选项超出屏幕时，会被遮挡
image

**Comment by yutao331763646** [comment]
修改之后：自动调整位置，使其剧中，和antd react体验保持一致，如下图
image

File: components/cascader/cascader.component.ts [hunk 1]
```diff
@@ -587,10 +587,15 @@ export class NzCascaderComponent implements NzCascaderComponentAsSource, OnInit,
...
     }
+
     this.el.focus();
     this.inSearchingMode
       ? this.cascaderService.setSearchOptionSelected(option as NzCascaderSearchOption)
       : this.cascaderService.setOptionActivated(option, columnIndex, true);
+    setTimeout(() => {
+      // 每次点击，等渲染完成（过渡时间），重新定位
+      this.overlay.overlayRef.updatePosition();
+    }, 10);
```
Comment (simplejason) [inline components/cascader/cascader.component.ts:598]: Avoid to use `setTimeout()`, check function `reposition` in this file

Comment (simplejason) [inline components/cascader/cascader.component.ts:598]: Avoid to use `setTimeout()`, check function `reposition` in this file

**Review by simplejason — CHANGES_REQUESTED** [review]

File: components/cascader/cascader.component.ts (hunk 1 above)
Comment (yutao331763646) [reply components/cascader/cascader.component.ts]: 我这个场景，是每次点击cascader-li时，右边的cascader-ul渲染出来，再重新调整overlay的位置，promise.resolve只能保证有新的变更时执行，但是我需要拿到变更的结果，所以用了detectChanges，同步跑了一次check，再更新位置 。@simplejason

File: components/cascader/cascader.component.ts [hunk 2]
```diff
@@ -587,7 +586,9 @@ export class NzCascaderComponent implements NzCascaderComponentAsSource, OnInit,
     if (option && option.disabled) {
       return;
     }
+
     this.el.focus();
+    console.log(this.inSearchingMode);
```
Comment (wendellhu95) [inline components/cascader/cascader.component.ts:591]: Please clean up console.log.

Comment (yutao331763646) [reply components/cascader/cascader.component.ts]: Sorry, everything is ok. @wendellhu95

**Review by simplejason — APPROVED** [review]
LGTM

**Pull request merged**